src/main/java/data/TaskManager.java

```diff
@@ -126,55 +129,55 @@ public List<Task> getTasksForDate(LocalDate date) {
     /**
      * Adds a task from user input along with the date.
      *
-     * @param scanner Scanner object to get user input.
      * @param weekView WeekView object to validate the date.
      * @param inMonthView A boolean indicating whether the view is in month view or not.
      * @throws TaskManagerException If there is an error in managing tasks.
      * @throws DateTimeParseException If there is an error parsing the date.
      */
-    public static void addManager(Scanner scanner, WeekView weekView, boolean inMonthView)
-            throws TaskManagerException, DateTimeParseException {
-        System.out.println("Enter the date for the task (dd/MM/yyyy):");
-        LocalDate date = parseInputDate(scanner);
+    public void addManager(WeekView weekView, MonthView monthView,boolean inMonthView, String action,
+            String day, String taskTypeString, String taskDescription)
+            throws TaskManagerException,DateTimeParseException {
 
-        if (inMonthView) {
-            checkIfDateInCurrentMonth(date);
+        // Convert the day to a LocalDate
+        LocalDate date;
+        // Convert the dayString to date range
+        String[] dates = new String[2];
+        if (day.contains("-")) {
+            dates = day.split("-");
         } else {
```

Comment:
This shorthand form of using days and hypens to add tasks quickly is sure to be useful for more experienced users